pkg/remote/aws/sqs_queue_details_fetcher.go

```diff
@@ -0,0 +1,43 @@
+package aws
+
+import (
+	"strings"
+
+	"github.com/cloudskiff/driftctl/pkg/resource"
+	"github.com/cloudskiff/driftctl/pkg/resource/aws"
+	"github.com/cloudskiff/driftctl/pkg/terraform"
+	"github.com/sirupsen/logrus"
+)
+
+type SQSQueueDetailsFetcher struct {
+	reader       terraform.ResourceReader
+	deserializer *resource.Deserializer
+}
+
+func NewSQSQueueDetailsFetcher(provider terraform.ResourceReader, deserializer *resource.Deserializer) *SQSQueueDetailsFetcher {
+	return &SQSQueueDetailsFetcher{
+		reader:       provider,
+		deserializer: deserializer,
+	}
+}
+
+func (r *SQSQueueDetailsFetcher) ReadDetails(res resource.Resource) (resource.Resource, error) {
+	ctyVal, err := r.reader.ReadResource(terraform.ReadResourceArgs{
+		ID: res.TerraformId(),
+		Ty: aws.AwsSqsQueueResourceType,
+	})
+	if err != nil {
```

Comment:
We should add a test to cover this case

Reply:
Yeah I'd like to but we don't have test for details fetcher with mock and I couldn't reproduce this one to have goldenfile...

Reply:
🆗